lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -673,11 +687,31 @@ public function computeChangeSet(ClassMetadata $class, $entity)
 
         // Look for changes in associations of the entity
         foreach ($class->getDeclaredPropertiesIterator() as $property) {
-            if (! ($property instanceof AssociationMetadata) || ($value = $property->getValue($entity)) === null) {
-                continue;
-            }
+            if ($property instanceof ToManyAssociationMetadata) {
+                $value = $property->getValue($entity);
+
+                if (null === $value) {
+                    continue;
+                }
+
+                $this->computeToManyAssociationChanges($property, $value);
+            } else {
```

Comment:
Didn't find a cleaner way for now. Also here, iterating over dedicated accessors would be better.

Reply:
`elseif ($property instanceof AssociationMetadata && !$isUnInitializedProxy)` :upside_down_face:

Reply:
Aye, but the `if` blocks still need de-duplication anyway